src/ansiblelint/config.py

```diff
@@ -164,5 +165,26 @@ def ansible_version(version: str = "") -> Version:
     return Version(version)
 
 
+def path_inject() -> None:
+    """Add python interpreter path to top of PATH to fix outside venv calling."""
+    # This make it possible to call ansible-lint that was installed inside a
+    # virtualenv without having to pre-activate it. Otherwise subprocess will
+    # either fail to find ansible executables or call the wrong ones.
+    #
+    # This must be run before we do run any subprocesses, and loading config
+    # does this as part of the ansible detection.
+    paths = [x for x in os.environ.get("PATH", "").split(os.pathsep) if x]
+    ansible_path = shutil.which("ansible")
+    if ansible_path:
+        ansible_path = os.path.dirname(ansible_path)
+    py_path = os.path.dirname(sys.executable)
+    # Determine if we need to manipulate PATH
+    if ansible_path not in paths or py_path != ansible_path:
+        paths.insert(0, py_path)
+        os.environ["PATH"] = os.pathsep.join(paths)
+        print(f"WARNING: PATH altered to include {py_path}", file=sys.stderr)
+
+
+path_inject()
```

Comment:
I don't like this hack of patching runtime stuff during import-time. Why can't it be the first thing called in the entry point?

Reply:
That is because we have lots of imports before that https://github.com/ansible-community/ansible-lint/blob/main/src/ansiblelint/__main__.py#L38 

You are welcomed to find a better way to do it in a follow-up. At least this change keeps a single place where we run code at import, not adding a new one.

Reply:
Following this approach makes the problem of relying on multiple unobvious side effects even worse than it is now. Also, I don't see why putting this in `main()` wouldn't work for the current PR.